crates/bevy_ecs/src/bundle.rs

```diff
@@ -405,11 +409,11 @@ impl BundleInfo {
                     }
                 }
                 StorageType::SparseSet => {
-                    sparse_sets.get_mut(component_id).unwrap().insert(
-                        entity,
-                        component_ptr,
-                        change_tick,
-                    );
+                    let sparse_set =
+                        // SAFETY: If component_id is in self.component_ids, the caller guarentees that
+                        // a sparse set exists for the component.
```

Comment:
These safety comments seem correct, but they doesn't really match the function's safety documentation:

> `table` must be the "new" table for `entity`. `table_row` must have space allocated for the `entity`, `bundle` must match this [`BundleInfo`]'s type

Can you change the function documentation to be more accurate? In particular it should mention how this applies to sparse set components. Seems like the callsites are all undocumented right now (oof) so it should be a simple change.

Reply:
I moved the invariant to `BundleInfo::new` (moved from `initialize_bundle`), where there's only one callsite.